internal/engine/experiment/tlsmiddlebox/testkeys.go

```diff
@@ -0,0 +1,46 @@
+package tlsmiddlebox
+
+import (
+	"sync"
+
+	"github.com/ooni/probe-cli/v3/internal/model"
+)
+
+// TestKeys contains the experiment results
+type TestKeys struct {
+	Queries        []*model.ArchivalDNSLookupResult  `json:"queries"`
+	TCPConnect     []*model.ArchivalTCPConnectResult `json:"tcp_connect"`
+	IterativeTrace []*CompleteTrace                  `json:"iterative_trace"`
+
+	mu sync.Mutex
+}
+
+// NewTestKeys creates new tlsmiddlebox TestKeys
+func NewTestKeys() *TestKeys {
+	return &TestKeys{
+		Queries:        []*model.ArchivalDNSLookupResult{},
+		TCPConnect:     []*model.ArchivalTCPConnectResult{},
+		IterativeTrace: []*CompleteTrace{},
+	}
+}
+
+// addqueries adds []*model.ArchivalDNSLookupResut to the test keys queries
+func (tk *TestKeys) addQueries(ev []*model.ArchivalDNSLookupResult) {
```

Comment:
Suggested pattern:

```
func (tk *TestKeys) addQueries(ev ...*model.ArchivalDNSLookupResult) {
```

This will simplify your life when you need to fill with a single element